rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,150 @@
+to the necessary files, depending on the SSH agent configuration. At a minimum this will allow the `teleport_ssh_service` domain to bind to
+the configured SSH port and read and write to the data directory, and dial the configured Teleport Proxy and Auth servers. 
+
+There are a few optional features that can be enabled or disabled on SSH agents, the first version of the policy will ensure necessary permissions
+for all optional features are allowed to simplify development and deployment.
+
+These include:
+
+- Reading `~/.tsh/environment` on the server before creating a session
+    - The `teleport_ssh_service_intermediary` domain will be allowed to read `.tsh/environment` in all users' home directories
+- Creation of host users and groups
+    - The `teleport_ssh_service` domain will be allowed to call necessary binaries such as `useradd`, `usermod` and `groupadd`
+- Enhanced session recording with BPF
+   - The `teleport_ssh_service` domain will be allowed to create and manage BPF ring buffers, and load BPF programs
+- Creation of PAM contexts
+    - The `teleport_ssh_service_intermediary` domain will be allowed to `dlopen` the PAM module and call it's exported functions
+    - The location of the PAM module will be configurable
+- TCP port forwarding
+    - The `teleport_ssh_service_networking` domain will be allowed to listen to arbitrary ports and connect to arbitrary network addresses
+- X11 forwarding
+    - The `teleport_ssh_service_networking` domain will be allowed to create and manage X11 sockets
+- SFTP support
+    - The `teleport_ssh_service_intermediary` domain will be allowed to re-execute itself with `sftp` as the first argument
+
+The SELinux policy will be created as a SELinux Reference Policy (refpolicy), to maximize readability and maintainability.
+
+The main alternative is to use SELinux Common Intermediate Language (CIL) which bridges refpolicies and compiled binary policy files.
+CIL is extremely powerful but much more difficult to understand by humans, it's meant for consumption by SELinux tools.
+
+#### Future updates
+
+The SELinux policy will be updated in the future to support new features and maintain compatibility with existing features.
+Any changes to the SSH service that result in new files being managed, new network activity (listening on new ports, dialing new addresses etc),
+new binaries being executed, or new Linux syscalls being called will require updates to the SELinux policy.
+
+### Installation and updating
+
+The policy source will be embedded in future Teleport agent binaries.
+
+A `selinux` subcommand will be added to Teleport agent binaries to allow users to install and configure the SELinux policy. This subcommand
+will use the Teleport agent configuration file to determine how to configure some of the tunable SELinux policy contexts, such as port
+numbers, file paths, and booleans using the `semanage` tool.
```

Comment:
This will add an external dependency to Teleport, that users will need installed on their machine if they want to use this feature. How will this be handled?

Reply:
You mean SELinux? This policy will be entirely opt in, if users don't have SELinux on their hosts that's out of scope for us to try to configure